File: cocmd_cli/core/models/script_model.py
from dataclasses import dataclass, field
from cocmd_cli.core.os import OS
from cocmd_cli.utils.io import DictLoader
from typing import List, Optional
from enum import Enum


class StepRunnerType(Enum):
    SHELL = "shell"
    MARKDOWN = "markdown"
    PYTHON = "python"
    COCMD_SCRIPT = "cocmd_script"
    LINK = "link"


@dataclass(frozen=True)
class StepModel(DictLoader):
    title: str
    runner: StepRunnerType
    content: str
    description: str


@dataclass(frozen=True)
class ScriptModel(DictLoader):
    steps: List[StepModel]
    env: Optional[OS] = field(default=OS.ANY)
    description: Optional[str] = field(default=None)

    def supports_os(self, os: OS) -> bool:
        return os in (self.env, OS.ANY)

File: cocmd_cli/src/core/models/script_model.rs
use serde_derive::{Serialize, Deserialize};
use crate::utils::sys::OS;

#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Hash, Clone)]
pub enum StepRunnerType {
    #[serde(alias="shell", alias="SHELL")]
    SHELL,
    #[serde(alias="markdown", alias="MARKDOWN")]
    MARKDOWN,
    #[serde(alias="python", alias="PYTHON")]
    PYTHON,
    #[serde(alias="link", alias="LINK")]
    LINK,
}

#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Hash, Clone)]
pub struct StepModel {
    pub runner: StepRunnerType,
    pub content: String,
    pub title: String,
}

#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Hash, Clone)]
pub struct ScriptModel {
    pub steps: Vec<StepModel>,
    pub env: Option<OS>,
    pub description: Option<String>,
}